sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -1472,6 +1475,8 @@ class RocksDB(
         log"Current lineage: ${MDC(LogKeys.LINEAGE, lineageManager)}")
       // Compare and update with the version that was just uploaded.
       lastUploadedSnapshotVersion.updateAndGet(v => Math.max(snapshot.version, v))
+      // Report snapshot upload event to the coordinator.
+      eventListener.foreach(_.reportSnapshotUploaded(snapshot.version))
```

Comment:
So we are reporting this even if changelog checkpointing is not being used? Is there a need for this if we are uploading snapshot for every batch (i.e. no changelog)?

Or Is it just for simplicity of the coordinator side logic for detection, since this is only specific to rocksdb? Just asking. I'm fine with just adding comment that it can be improved, if we are doing this for now, just to not overly complicate things

Reply:
Yeah this is tricky, because the coordinator needs a way to distinguish between the two cases {all partitions are lagging} and {messages are disabled}, and the checks needed for this is different between RocksDB and HDFS stores because of the changelog config.

I'll add a short comment to explain the nuances behind it :+1: